crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -602,7 +602,7 @@ impl<P: RoomDataProvider, D: Decryptor> TimelineController<P, D> {
     pub(super) async fn subscribe_filter_map<U, F>(
         &self,
         f: F,
-    ) -> (Vector<U>, impl Stream<Item = VectorDiff<U>>)
+    ) -> (Vector<U>, impl Stream<Item = VectorDiff<U>> + use<U, F, P, D>)
```

Comment:
Same here. I believe `impl + use` creates more confusion that using a named type. In case there isn't a named type, I think it's great, but since we've one, can't we use it?